app/views/courses/TeacherClassView.coffee

```diff
@@ -0,0 +1,176 @@
+    @progressDotTemplate = require 'templates/courses/progress-dot'
+    
+    @sortAttribute = 'name'
+    @sortDirection = 1
+    
+    @classroom = new Classroom({ _id: classroomID })
+    @classroom.fetch()
+    @supermodel.trackModel(@classroom)
+    
+    @listenTo @classroom, 'sync', ->
+      @students = new Users()
+      @students.fetchForClassroom(@classroom)
+      @supermodel.trackCollection(@students)
+      @listenTo @students, 'sync', @sortByName
+      @listenTo @students, 'sort', @render
+      
+      @classroom.sessions = new LevelSessions()
+      @classroom.sessions.fetchForAllClassroomMembers(@classroom)
+      @supermodel.trackCollection(@classroom.sessions)
+      
+    @courses = new Courses()
+    @courses.fetch()
+    @supermodel.trackCollection(@courses)
+    
+    @campaigns = new Campaigns()
+    @campaigns.fetchByType('course')
+    @supermodel.trackCollection(@campaigns)
+    
+    @courseInstances = new CourseInstances()
+    @courseInstances.fetchByOwner(me.id)
+    @supermodel.trackCollection(@courseInstances)
+    
+
+  onLoaded: ->
+    console.log("loaded!")
+    
+    @classCode = @classroom.get('codeCamel') || @classroom.get('code')
+    @joinURL = document.location.origin + "/courses?_cc=" + @classCode
+    
+    @earliestIncompleteLevel = helper.calculateEarliestIncomplete(@classroom, @courses, @campaigns, @courseInstances, @students)
+    @latestCompleteLevel = helper.calculateLatestComplete(@classroom, @courses, @campaigns, @courseInstances, @students)
+    for student in @students.models
+      # TODO: this is a weird hack
+      studentsStub = new Users([ student ])
+      student.latestCompleteLevel = helper.calculateLatestComplete(@classroom, @courses, @campaigns, @courseInstances, studentsStub)
+      
+    classroomsStub = new Classrooms([ @classroom ])
+    @progressData = helper.calculateAllProgress(classroomsStub, @courses, @campaigns, @courseInstances, @students)
+    super()
+    
+  copyCode: ->
+    @$('#join-code-input').val(@classCode).select()
+    @tryCopy()
+  
+  copyURL: ->
+    @$('#join-url-input').val(@joinURL).select()
+    @tryCopy()
+    
+  tryCopy: ->
+    try
+      document.execCommand('copy')
+      application.tracker?.trackEvent 'Classroom copy URL', category: 'Courses', classroomID: @classroom.id, url: @joinURL
+    catch err
+      console.log('Oops, unable to copy', err)
+
+  onClickAddStudents: (e) =>
+    modal = new InviteToClassroomModal({ classroom: @classroom })
+    @openModalView(modal)
+    @listenToOnce modal, 'hide', @render
+    
+  onClickEnrollStudent: (e) ->
+    # TODO
+  
+  sortByName: (e) =>
+    if @sortValue == 'name'
+      @sortDirection = -@sortDirection
+    else
+      @sortValue = 'name'
+      @sortDirection = 1
+      
+    dir = @sortDirection
+    @students.comparator = (student1, student2) ->
+      return (if student1.get('name') < student2.get('name') then -dir else dir)
+    @students.sort()
+    
+  sortByProgress: (e) =>
```

Comment:
`=>`'s unnecessary for both `sortByName` and `sortByProgress`. Backbone `events` and `listenTo` bind to the view by default.